test/hotspot/gtest/runtime/test_classPrinter.cpp

```diff
@@ -51,16 +53,16 @@ TEST_VM(ClassPrinter, print_methods) {
   stringStream s1;
   ClassPrinter::print_methods("*ang/Object*", "wait", 0x1, &s1);
   const char* o1 = s1.freeze();
-  ASSERT_TRUE(strstr(o1, "class java/lang/Object loader data:") != NULL) << "must find java/lang/Object";
-  ASSERT_TRUE(strstr(o1, "method wait : (J)V")    != NULL) << "must find java/lang/Object::wait(long)";
-  ASSERT_TRUE(strstr(o1, "method wait : ()V")     != NULL) << "must find java/lang/Object::wait()";
-  ASSERT_TRUE(strstr(o1, "method finalize : ()V") == NULL) << "must not find java/lang/Object::finalize";
+  ASSERT_THAT(o1, HasSubstr("class java/lang/Object loader data:")) << "must find java/lang/Object";
+  ASSERT_THAT(o1, HasSubstr("method wait : (J)V")) << "must find java/lang/Object::wait(long)";
+  ASSERT_THAT(o1, HasSubstr("method wait : ()V")) << "must find java/lang/Object::wait()";
+  ASSERT_THAT(o1, Not(HasSubstr("method finalize : ()V"))) << "must not find java/lang/Object::finalize";
```

Comment:
It required some research for me to understand how `Not` is working here and below.  It's relying on ADL.
That used to be at best unreliable or not work at all in some common compilers, but I think that's mostly
fixed now.  Since we don't use namespaces in HotSpot, ADL is not encountered much in our code.  I'm
okay with this, but others might be surprised too.